internal/registry/service.go

```diff
@@ -5,15 +5,43 @@ import (
 	"errors"
 	"fmt"
 
-	packerSvc "github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/client/packer_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/client/packer_service"
 	"github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/models"
 	"google.golang.org/grpc/codes"
 )
 
+func (client *Client) CreateBucket(
```

Comment:
I'm a bit confused and might be missing some context here, but it looks like you are adding methods to the client that already exists as functions. Is there a particular reason for this as opposed to testing the functions as they are?

Reply:
When I created my original PR and where I missed the current existing code, that's how I implemented it. I think doing it this way makes more sense, so when I moved my package over I left those. 🤔

Reply:
Okay cool, thanks for that context. It's def. important that the code makes sense to the reader. Instead of duplicating the code let's pair up and talk about how we can clear up the existing code. 

FWIW it was a conscious decision to write the functions as I did but I realize that I didn't add comments on the rationale behind it, which probably would've helped . We should talk through the rationale a bit and address where necessary.